vminfo_parser/analyzer.py

```diff
@@ -307,129 +285,105 @@ def sort_by_disk_space_range(
         """
+        df = self.vm_data.create_environment_filtered_dataframe(
+            self.config.environments, self.config.environment_filter
+        )
+
+        if os_filter:
+            df = df[df["OS Name"] == os_filter]
+
         diskHeading = self.vm_data.column_headers["vmDisk"]
         disk_space_ranges = self.calculate_disk_space_ranges(
-            dataFrame=dataFrame,
-            show_disk_in_tb=show_disk_in_tb,
-            over_under_tb=over_under_tb,
+            dataFrame=df,
+            show_disk_in_tb=self.config.breakdown_by_terabyte,
+            over_under_tb=self.config.over_under_tb,
         )
 
         for lower, upper in disk_space_ranges:
-            mask = (dataFrame[diskHeading] >= lower) & (dataFrame[diskHeading] <= upper)
-            dataFrame.loc[mask, "Disk Space Range"] = f"{lower}-{upper} GB"
-
-        if environment_filter is None:
-            environment_filter = "all"
-
-        if granular_disk_space_by_os:
-            sorted_range_counts_by_environment = self.sort_by_disk_space_range(
-                dataFrame,
-                drop_columns=["second_number", "OS Name"],
-                os_breakdown=True,
-                environment_filter=environment_filter,
-            )
-        else:
-            sorted_range_counts_by_environment = self.sort_by_disk_space_range(
-                dataFrame,
-                drop_columns=["second_number"],
-                os_breakdown=False,
-                environment_filter=environment_filter,
-            )
-
-        if not sorted_range_counts_by_environment.empty:
-            # Now call the print_formatted_disk_space method
-            self.cli_output.print_formatted_disk_space(
-                sorted_range_counts_by_environment,
-                os_filter=os_filter,
-            )
-
-        # Call the new visualize method
-        if self.config.generate_graphs:
-            if environment_filter == "all":
-                self.visualizer.visualize_disk_space_horizontal(sorted_range_counts_by_environment)
-            else:
-                self.visualizer.visualize_disk_space_vertical(
-                    sorted_range_counts_by_environment,
-                    os_filter=os_filter,
-                )
+            mask = (df[diskHeading] >= lower) & (df[diskHeading] <= upper)
+            df.loc[mask, "Disk Space Range"] = f"{lower}-{upper} GB"
 
-    def handle_operating_system_counts(self: t.Self, environment_filter: str, dataFrame: pd.DataFrame = None) -> None:
-        """Handles the counting of operating systems based on the provided environment filter.
+        return self.sort_by_disk_space_range(df)
 
-        This function calculates the counts of operating systems and outputs the results.
-        It also visualizes the distribution of operating systems if the configuration allows for it.
+    def get_unique_os_names(self: t.Self, df: pd.DataFrame | None = None) -> list[str]:
```

Comment:
Im actually going to remove it from the signature from now.  If we find a need for it later,  its trivial to put back in,  but since its not used today,  I would rather remove it,  so we dont have to wonder later why it is there at all